### PR TITLE
feat: add visible_on_all_workspaces, closes #6589

### LINKED ISSUE #6589: [feat] Can Tauri be visible in all workspaces of macOS?
### Describe the problem

CleanShot 2023-03-29 at 21 39 37

I've searched for a lot of ways to implement a window that shows in all workspaces when switching workspaces in macOS, similar to the settings in Electron, as in the screenshot below, but I haven't found a way to do it, and I wonder if there's another way to do it?
CleanShot 2023-03-29 at 21 28 22

### Describe the solution you'd like

Is it possible to add this setting, or is there another solution?

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by krzykro2**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information
It's my first PR to Tauri, please be gentle. I basically added `visible_on_all_workspaces` to all the places where I found `always_on_top`. Haven't found any tests for `always_on_top` so haven't added any for my feature. Tested locally on my Linux and works great (given the Tao implementation I would assume it should work great on Mac also).

### CODE REVIEW

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks for contribution, could you run `cargo fmt` and add a change file in `.changes`?

**Comment by krzykro2** [comment]
> Thanks for contribution, could you run `cargo fmt` and add a change file in `.changes`?

Done. PTAL.

File: core/tauri/src/window.rs [hunk 1]
```diff
@@ -1479,6 +1488,18 @@ impl<R: Runtime> Window<R> {
       .map_err(Into::into)
   }
 
+  /// Determines if this window should be visible on all workspaces or virtual desktops.
```
Comment (amrbashir) [inline core/tauri/src/window.rs]:
```suggestion
  /// Sets whether the window should be visible on all workspaces or virtual desktops.
```

File: core/tauri-utils/src/config.rs [hunk 2]
```diff
@@ -962,6 +962,9 @@ pub struct WindowConfig {
   /// Whether the window should always be on top of other windows.
   #[serde(default, alias = "always-on-top")]
   pub always_on_top: bool,
+  /// Whether the window should be visible on all workspaces.
```
Comment (amrbashir) [inline core/tauri-utils/src/config.rs]:
```suggestion
  /// Whether the window should be visible on all workspaces or virtual desktops.
```

File: core/tauri-runtime/src/webview.rs [hunk 3]
```diff
@@ -248,6 +248,10 @@ pub trait WindowBuilder: WindowBuilderBase {
   #[must_use]
   fn always_on_top(self, always_on_top: bool) -> Self;
 
+  /// Whether the window should be visible on all workspaces.
```
Comment (amrbashir) [inline core/tauri-runtime/src/webview.rs]:
```suggestion
  /// Whether the window should be visible on all workspaces or virtual desktops.
```

File: .changes/add-visible-on-all-workspaces.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+"tauri": 'minor:feat'
+"tauri-runtime": 'minor:feat'
+"tauri-utils": 'minor:feat'
+---
+
+Added `visible_on_all_workspaces` configuration option to `WindowBuilder`,
+`Window`, and `WindowConfig`.
```
Comment (amrbashir) [inline .changes/add-visible-on-all-workspaces.md]:
```suggestion
---
"tauri": 'minor:feat'
"tauri-runtime": 'minor'
"tauri-utils": 'minor:feat'
---

Added `visible_on_all_workspaces` configuration option to `WindowBuilder`, `Window`, and `WindowConfig`.
```

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks

**Review by amrbashir — DISMISSED** [review]

**Comment by amrbashir** [comment]
Oh looks like you didn't sign some of your commits and I can't merge the PR unless they are signed. You need to setup [commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits), then you can sign past commit like [this](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits/1123928#1123928) for example.

File: core/tauri-config-schema/schema.json [hunk 5]
```diff
@@ -445,6 +445,11 @@
           "default": false,
           "type": "boolean"
         },
+        "visibleOnAllWorkspaces": {
+          "description": "Whether the window should be visible on all workspaces.",
```
Comment (amrbashir) [inline core/tauri-config-schema/schema.json]:
```suggestion
          "description": "Whether the window should be visible on all workspaces or virtual desktops.",
```

File: tooling/cli/schema.json [hunk 6]
```diff
@@ -445,6 +445,11 @@
           "default": false,
           "type": "boolean"
         },
+        "visibleOnAllWorkspaces": {
+          "description": "Whether the window should be visible on all workspaces.",
```
Comment (amrbashir) [inline tooling/cli/schema.json]:
```suggestion
          "description": "Whether the window should be visible on all workspaces or virtual desktops.",
```

**Comment by krzykro2** [comment]
I might have failed at git :facepalm: 

I managed to successfully sign my old commit by rebasing back onto dev, however your commit seems to be unsigned now?

File: core/tauri-config-schema/schema.json (hunk 5 above)
Comment (amrbashir) [inline core/tauri-config-schema/schema.json]:
```suggestion
          "description": "Whether the window should be visible on all workspaces or virtual desktops.",
```

File: tooling/cli/schema.json (hunk 6 above)
Comment (amrbashir) [inline tooling/cli/schema.json]:
```suggestion
          "description": "Whether the window should be visible on all workspaces or virtual desktops.",
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — APPROVED** [review]

**Pull request merged**